### ISSUE TITLE
Feature request: Add suggestion(s) to error messages

### ISSUE THREAD

**Issue by iykekings** [denoland/deno_lint#162]
Error messages should also have suggestions for possible fixes of the error.

**Comment by bartlomieju** [comment]
This would be a very useful feature to have. 

It can be implemented by adding a new field to `LintDiagnostic` called `suggestion` or `hint`, which could optionally be set when creating diagnostics in rules.

```rust
pub struct LintDiagnostic {
  pub range: Range,
  pub filename: String,
  pub message: String,
  pub code: String,
  pub suggestion: Option<String>,
}
```

`Option<String>` should be a valid choice because not all rules must provide suggestions (eg. in trivial rules it might not make sense to add suggestions.

**Comment by iykekings** [comment]
@bartlomieju could there be a scenario where there are multiple solutions to one error?

**Comment by bartlomieju** [comment]
@iykekings sorry I missed you comment. Yeah, there might be such scenario - in that case we can change `LintDiagnostic::hint` to be `Vec<String>`.

**Comment by jcc10** [comment]
> (eg. in trivial rules it might not make sense to add suggestions.)

I would just like to add that trivial depends on who you are talking to. Some cases may be trivial but the conventional way is the wrong way.

The best example is ```const self = this```. This is a "hack" for when you are running a function in a different scope and the linter hates it. But it has been the convention for the past forever. The better(?) way is to use a arrow function, but even I, who found this out over a year ago, will still look up if I am remembering that fact correctly.

**Comment by bartlomieju** [comment]
I'm going to close this issue now, most of the rules now include a `hint` suggesting what to change to make linter happy.
